updating/update-using-custom-machine-config-pools.adoc

```diff
@@ -0,0 +1,167 @@
+
+[NOTE]
+----
+Creating custom MCP using master nodes is not suported.
+Machine config operator (responsible for updating the nodes) will ignore the custom MCP using the master nodes.
+This is required to make sure control plane nodes remain stable.
+----
+
+An OpenShift or Kubernetes cluster is highly available by design and there are many Kuberenetes features (e.g. pod disruption budget, pod affinity, health checks, replicas) which can make the application highly available in case of node failures.
+
+However there might be some secenarios where a controlled rollout of upgrade to the worker nodes is desired. This helps to make sure the mission critical application stays available during the whole update process even if the upgrade process cause a failure. It is also useful in scenarios where we can not have big maintaine window to complete the whole update process. This is can be characterized as a canary release i.e. we can control the rollout of worker nodes by controlling the machine config pools. After the update of the first MCP, the application compatibility can be verfied and then the whole fleet can be updated gradually to the new version.
+
+[NOTE]
+----
+A node can be part of at most one MCP. The machine config operator rolls out update to all the unpaused worker pools simultaneously.
+For example, if maxUnavailable is set to one then 1 node from each MCPs will be updated concurrently.
+
+During the update process of nodes (as per are the  `maxUnavailable` value) in a MCP are cordoned and drained before it is updated.
+Once the update process completes the nodes are un-cordned.
+----
+
+== Workflow of doing a canary release using MCP
+
+. Create MCPs out of the worker pool. The number of nodes in each MCP depends on few factors like maintaince window duration for each MCP, the amount of reserve capacity (extra worker nodes) present.
++
+[NOTE]
+====
+In case of an failure i.e. the MCP with the new version does not work as expected the applications, the extra capacity will help to maintain the quality of service of the applications.
+====
++
+. Pause the MCPs created in the previous step.
++
+. Start the update process. The update process will only update the MCP which contains the master nodes i.e. the control plane.
++
+. Once the control plane update is completed, unpause one MCP. It will update the MCP to the new version of OCP. You can check the progress of the update in the web console (In Administrator view -> Administration -> Cluster settings ) as well as running `$ oc get mcp` CLI command.
```

Comment:
Doesn't have to be a "new version of OCP"; this process could apply to any MachineConfig update.  OpenShift updates are one source of MachineConfig updates, but they are not the only source.  For example, users may also create new MachineConfigs directly, or indirectly via KubeletConfig.

I also usually don't include `$` prompts in commands unless I'm creating them in a block, so probably:

> ... running the `oc get mcp` CLI command.

or some such for your inline command.

And also, no need to abbreviate here; maybe `oc get machineconfigpools.machineconfiguration.openshift.io`.  Not sure what the doc precedent is for that sort of thing...

Reply:
Fixed in https://github.com/openshift/openshift-docs/commit/c27e14a1b4785cc8704f11d191797f198c10f027